### PR TITLE
[FLINK-12024] Bump universal Kafka connector to Kafka dependency to 2.2.0

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request bumps universal Kafka connector to Kafka dependency to 2.2.0*

## Brief change log

  - *Bump universal Kafka connector to Kafka dependency to 2.2.0*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by yanghua** [comment]
cc @aljoscha and @pnowojski

**Comment by zentol** [comment]
No problems in terms of licensing.

**Comment by bolkedebruin** [comment]
Please, this fixes some really annoying bugs in Kafka

**Comment by eliaslevy** [comment]
While the universal tries to track the latest Kafka client version, at times it may fall behind.  E.g. before this update it was using 2.0.1, skipping the 2.1.x series entirely.  At the moment the version of the client used for the universal connector is not mentioned in the Flink documentation.  It may be useful to do so.  Users may be interested is knowing the specific version so they can tell whether specific features or fixes are included in the connector.

**Comment by yanghua** [comment]
@eliaslevy Your comment makes sense. I agree with you. I suggest we should describe the current kafka client version which the universal kafka connector tracks in the documentation. Any time when we upgrade the kafka's version, we also should update the document. What do you think? @pnowojski @aljoscha @zentol

**Comment by aljoscha** [comment]
I think this makes sense, yes.

**Comment by yanghua** [comment]
@aljoscha OK, thanks for your reply. I will remark the kafka client version in the document.

File: docs/dev/connectors/kafka.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ For most users, the `FlinkKafkaConsumer08` (part of `flink-connector-kafka`) is
         <td>1.7.0</td>
         <td>FlinkKafkaConsumer<br>
         FlinkKafkaProducer</td>
-        <td>>= 1.0.0</td>
+        <td>>= 1.0.0 (until 2.2.0)</td>
```
Comment (yanghua) [inline docs/dev/connectors/kafka.md]: @aljoscha and @pnowojski What do you think about this change?

**Comment by eliaslevy** [comment]
That's no good, because the Kafka client is also forward compatible, except
for be features. An that is needed is just a comment stating the version
used.

**Comment by yanghua** [comment]
@eliaslevy Sounds reasonable. My change will make the users confused.

File: docs/dev/connectors/kafka.md [hunk 2]
```diff
@@ -88,7 +88,7 @@ For most users, the `FlinkKafkaConsumer08` (part of `flink-connector-kafka`) is
         <td>>= 1.0.0</td>
         <td>
         This universal Kafka connector attempts to track the latest version of the Kafka client.
-        The version of the client it uses may change between Flink releases.
+        The version of the client it uses may change between Flink releases. Now, it uses Kafka client 2.2.0.
```
Comment (yanghua) [inline docs/dev/connectors/kafka.md]: @eliaslevy What about this change?

**Comment by eliaslevy** [comment]
Good enough, but I would rephrase it as "As of this release, it uses the Kafka 2.2.0 client."

**Comment by yanghua** [comment]
@eliaslevy Thanks for your suggestion, accepted your opinion.

**Comment by yanghua** [comment]
@pnowojski Thanks for pointing these version number. I have upgraded them to `2.2.0` except :
`flink-filesystems/flink-fs-hadoop-shaded/src/main/resources/META-INF/NOTICE:Azure Data Lake Store - Java client SDK 2.0.11`.

Moreover, I found the first arg `KAFKA_CONNECTOR_VERSION` in `test_sql_client_kafka_common.sh` seems useless. I suggest that we should remove them in another  PR in the future.

File: flink-end-to-end-tests/test-scripts/test_sql_client.sh [hunk 3]
```diff
@@ -20,7 +20,7 @@
 set -Eeuo pipefail
 
 KAFKA_CONNECTOR_VERSION="2.0"
```
[the remark this answers is not in the document]
Comment (yanghua) [reply flink-end-to-end-tests/test-scripts/test_sql_client.sh]: OK, I will drop it in a separate commit.

[the remark this answers is not in the document]
Comment (yanghua) [reply flink-end-to-end-tests/test-scripts/test_sql_client.sh]: @pnowojski I have finished this work, can you review again? thanks.
